### PR TITLE
Controllers implementation completed.

### PR DESCRIPTION
**Pull request by ometman**
I have completed the implementation of controllers in branch 'controllers':
- Rails generate scaffold command was not used
- The url of the Wireframes was used.
- For each URL a Route,  Correct Controller and View file was create.
- The view files contain only placeholders
- Code was added handle only the 4 required URLs.
- Linters checked and passed.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -8,4 +12,7 @@
 
   # Defines the root path route ("/")
   # root "posts#index"
+  resources :users, only: [:index, :show] do
+    resources :posts, only: [:index, :show]
```
Comment (topeogunleye) [inline config/routes.rb:4]: Your route definitions are well-structured, and you've successfully implemented the nested resources for users and posts, which aligns with the project requirements. However, it seems there are some additional routes present beyond the four required URLs specified in the project guidelines.

To meet the project requirements, ensure that your routes are limited to only the following:

```ruby
resources :users, only: [:index, :show] do
  resources :posts, only: [:index, :show]
end
```

It's important to have a clean and minimalistic routing structure to adhere to project specifications. Please review your routes and remove any that are not explicitly required. If there are specific routes necessary for other functionalities, make sure they align with the project guidelines.

Feel free to reach out if you have any questions or need further clarification. Keep up the good work! 👍🚀

**Review by topeogunleye — CHANGES_REQUESTED** [review]
Hi @ometman, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

Try-Again

> ## (Highlights) Good Points: 💪 
**Code Review Highlights:**

1. **Manual Controller Setup:**
   - Excellent job not relying on the `rails generate scaffold` command. Building controllers from scratch showcases a deeper understanding of Rails architecture and provides a hands-on experience in setting up controllers independently.

2. **Controller Actions:**
   - Well-organized `UsersController` and `PostsController` with the required `#index` and `#show` actions, demonstrating a clear understanding of RESTful conventions.

3. **Empty Actions Placeholder:**
   - Thoughtful inclusion of placeholder actions with the comment `def index; end` and `def show; end`. This indicates a proactive approach, acknowledging that these actions will be fleshed out as the project progresses.

4. **View Files:**
   - Implementation of corresponding view files for each controller action (`index` and `show`) is spot on. This ensures a structured separation of concerns and paves the way for future development.

5. **Project Adherence:**
   - Your adherence to the project requirements is commendable. By manually setting up controllers and view files, you're maximizing the learning experience intended for this project.

6. **Code Structure:**
   - The code structure is clean and follows Rails conventions, promoting maintainability and readability.

7. **Future-Ready Setup:**
   - Although the actions are currently empty, having them in place sets a solid foundation for future functionality, making your codebase scalable.

Keep up the excellent effort! 🌟🚀

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Ghiftee — APPROVED** [review]
## STATUS: APPROVED ♻️♻️
Hi @ometman, 

To highlight 👍🏽 
✔️ Your app runs as expected
✔️ Route structure is clean
✔️ No linter errors

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀  
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ometman** [comment]
Thanks

**Pull request merged**